documentation/IntegrationProvider.md

```diff
@@ -87,9 +90,52 @@ showing the popup. An example of such invocation could be:
 setSwitchboardConfig({origin: 'acme', title: 'ACME Switchboard'});
 ````
 
+## Preflight API
+
+Sometimes the resource provider displays the Switchboard popup only for the
+user to discover that there are no matching tools for that particular resource.
+To avoid such cases the Switchboard offers the preflight API, which can be used
+to test how many tools a particular resource can match.
+
+When integrating the Switchboard popup, the resource provider can call the
+`testSwitchboardMatches(arrayOfLinksToResources)` javascript function. This
+function returns a promise that resolves to a json object with two keys:
+
+- `timeout`: specifies if a timeout has been triggered
+- `matches`: specifies how many tools the Switchboard matched with the
+specified resource(s).
+
+This function can be invoked after the page loads and its results used to,
+for example, disable the buttons that would display the Switchboard popup
+for some resources; or it can be invoked when the user opens a menu with a
+button that would display the Switchboard popup. However, care must be taken
+so that the users are not unpleasantly surprised by a sudden change in the page
+that would break their motions.
+
+An example of such an invocation would be:
+
+html```
+<li>
+<a href="javascript:;"
+  onclick="testSwitchboardMatches(['http://hdl.handle.net/1234/567890']).then((data) => {
+    if (data.timeout) return;
+    const li = this.parentElement.querySelector('ul li');
+    if (data.matches === 0) li.setAttribute('class', 'disabled');
+    let text = 'Process with Switchboard ';
+    if (data.matches == 0) text += '(no matches)';
+    else if (data.matches == 1) text += '(one match)';
+    else text += `(${data.matches} matches)`;
+    li.querySelector('a span').textContent = text;
+  });">
+    <i class="glyphicon glyphicon-open-file" aria-hidden="true"></i>
+    <span>Switchboard</span>
+</a>
+</li>
+```
```

Comment:
doesn't render, at least not at https://github.com/clarin-eric/switchboard-doc/blob/d4883306a03554bd5fc62170bddbab77b72996f2/documentation/IntegrationProvider.md